build_tools/third_party/mpi/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+find_package(MPI REQUIRED)
+
+external_cc_library(
+  PACKAGE
+    mpi
+  NAME
+    headers_and_libs
+  ROOT
+    "${IREE_SOURCE_DIR}/third_party/mpi/"
```

Comment:
This should be derived from where the packages was found. Installing MPI using distro packages will not place the installation inside the IREE tree.

Unfortunately, FindMPI.cmake only uses `MPI_HOME` and does not export a root directory. A root can be inferred though from the include or library path.